lib/fast_jsonapi/object_serializer.rb

```diff
@@ -176,9 +175,23 @@ def set_id(id_name = nil, &block)
       end
 
       def cache_options(cache_options)
-        self.cached = cache_options[:enabled] || false
-        self.cache_length = cache_options[:cache_length] || 5.minutes
-        self.race_condition_ttl = cache_options[:race_condition_ttl] || 5.seconds
+        cache_options = cache_options.dup
+
+        if cache_options.key?(:store)
+          self.cache_store_instance = cache_options.delete(:store)
+        elsif !cache_store_instance
+          warn('DEPRECATION WARNING: `store:` is required, we will default to `Rails.cache`.')
+          self.cache_store_instance =  Rails.cache
+        end
+
+        %i[enabled cache_length].each do |key|
+          if cache_options.key?(key)
+            warn("DEPRECATION WARNING: `#{key}:` is a deprecated cache option and has no effect anymore.")
+            cache_options.delete(key)
+          end
+        end
+
+        self.cache_store_options = (cache_store_options || {}).merge(cache_options)
```

Comment:
I'm not sure we need this and the initial `cache_options = cache_options.dup`. Plus merging the configuration every time might cause mutation problems on inherited serializer classes. Consider the case where one serializer inherits the other one but it has different cache options, calling `merge()` will carry over anything that was not overwritten.

Let's just keep `self.cache_store_options = cache_options.dup` to avoid any unexpected behavior for now.

Reply:
The initial dup might not be necessary but it doesn't hurt either. It only makes sure the hash is not modified ... who knows whether users pass some static configuration hash as an option, then we shouldn't delete keys from it. It is just some safety measure.

> Consider the case where one serializer inherits the other one but it has different cache options, calling merge() will carry over anything that was not overwritten.

I'd say exactly this is how it should be, or not? It inherits every option but can change it, otherwise what is inherited goes. This is how it was before, too.

Otherwise I must always pass a `store:`, too, if I call `cache_options`. Which makes me understand your comment above better about the `elsif`.

So you think you should have to pass *all* options on every call to `cache_options`, even on inheritance?

Reply:
@doits yep, I agree, let's be explicit, or we risk getting in trouble, let's reset the `cache_store_options` on every call to avoid surprises.

Reply:
I don't think it is a good design choice from the point of inheritance, but I want to have it finished, too, so I implemented it.